plugins/modules/aci_l4l7_service_graph_template_node.py

```diff
@@ -0,0 +1,359 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, Tim Cragg (@timcragg)
+# Copyright: (c) 2025, Shreyas Srish (@shrsr)
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: aci_l4l7_service_graph_template_node
+short_description: Manage L4-L7 Service Graph Templates Nodes (vns:AbsNode)
+description:
+- Manage Manage L4-L7 Service Graph Templates Nodes.
+options:
+  tenant:
+    description:
+    - The name of an existing tenant.
+    type: str
+    aliases: [ tenant_name ]
+  service_graph:
+    description:
+    - The name of an existing Service Graph.
+    type: str
+  node:
+    description:
+    - The name of the Service Graph Template Node.
+    type: str
+  func_template_type:
+    description:
+    - The functional template type for the node.
+    - The APIC defaults to C(other) when unset during creation.
+    type: str
+    choices: [ fw_trans, fw_routed, adc_one_arm, adc_two_arm, other ]
+  func_type:
+    description:
+    - The type of connection.
+    - The APIC defaults to C(go_to) when unset during creation.
+    type: str
+    choices: [ go_to, go_through, l1, l2 ]
+  device:
+    description:
+    - The name of an existing logical device.
+    type: str
+  device_tenant:
+    description:
+    - The tenant the logical device exists under.
+    - This variable is only used if logical device and node exist within different tenants.
+    - Intended use case is when the device is in the C(common) tenant but the node is not.
+    type: str
+  managed:
+    description:
+    - Whether this device managed by the apic.
+    - The APIC defaults to C(true) when unset during creation.
+    type: bool
+  routing_mode:
```

Comment:
has options specified in model, should we restrict to those choices?

Reply:
ok

Reply:
Is one of the options a default?